### PR TITLE
Update Trustless Swap guide

### PR DESCRIPTION
**Pull request by wriches**
## Description 

Updated the trustless swap guide to a new reference format

## Test plan 

Local + Vercel preview

### CODE REVIEW

File: docs/content/guides/developer/app-examples/trustless-swap.mdx [hunk 1]
```diff
@@ -1,27 +1,1777 @@
...
+		return res.status(400).send(e);
+	}
+});
+```
+
+### Next steps
+
+With the code successfully deployed on Testnet, you can now [create a frontend](#frontend) to display the trading data and to allow users to interact with the Move modules.
+
+## Frontend
```
Comment (amnn) [inline docs/content/guides/developer/app-examples/trustless-swap.mdx:958]: This section is starting to read a little like annotated code, rather than a guide, again. It's describing what every part of the finished codebase does, rather than how you build up the functionality to get to the end state.

For example, the first section talks about setting up providers -- the impression I would get from reading  that is that I should copy paste that code in, but in fact most of that code comes from running `pnpm create @mysten/dapp --template react-client-dapp`, along with setting up other things (like a `package.json`, etc) which we also haven't mentioned here.

I suspect that this is inevitable if you only have access to the final artifact and not the person who went through the journey of building it. I notice that in my own wording for how the indexer part works as well (above), where I'm not aware of the commands we might use to include prisma as a dependency, or tweak its schemas, but I can understand the output.

TL;DR we should involve @manolisliolios here and get him to explain the broad strokes of the journey.

Comment (amnn) [reply docs/content/guides/developer/app-examples/trustless-swap.mdx:969]: I was thinking the latter -- meeting with @wriches to explain more of the journey, i.e.

- Steps taken (the important thing here is to call out the order in which you did things, and when there's something that was generated, rather than written, what command was used to do that, etc).
- Non-obvious decisions and why you made them (particularly if it avoids a pitfall).
